src/TestExplorer/TestExplorer.ts

```diff
@@ -147,18 +149,27 @@ export class TestExplorer {
             if (isPathInsidePath(uri.fsPath, folder.folder.fsPath)) {
```

Comment:
Do we want to run this on every document symbols response, or when the active file is saved?
EDIT: although I see what you are doing, you are falling back to document symbols if LSP fails

Reply:
I think we want to run this on every response, as I'll often be typing out a new test declaration and it will show up immediately instead of waiting until save.

And yep, if `getDocumentTests` fails because the LSP isn't supported we fall back to the old way.

Reply:
I guess `textDocument/tests` shouldn't be too heavy a request for the LSP server